### ISSUE TITLE
Cannot activate block styles via keyboard

### ISSUE THREAD

**Issue by ocean90** [WordPress/gutenberg#10908]
**Describe the bug**
After focusing a block style and pressing the Enter key nothing happens.

**Expected behavior**
Just like selecting another block type via Enter, the new style should get applied when pressing the Enter key.

**Additional context**
The `jsx-a11y/click-events-have-key-events` rule has been disabled. https://github.com/WordPress/gutenberg/blob/1960a48e26ba949723e5a304cc1c79780c7085dd/packages/editor/src/components/block-styles/index.js#L90

**Comment by ocean90** [comment]
@aduth Right, I tried with a button first but funny enough, I had a custom block with a button so that didn't really work out. Added a possible solution in #10909.
